kernel/Kconfig.device

```diff
@@ -33,6 +33,15 @@ config DEVICE_DT_METADATA
 	  each device. This allows you to use device_get_by_dt_nodelabel(),
 	  device_get_dt_metadata(), etc.
 
+config DEVICE_PROVIDE_DEINIT
+	bool "Provide a mean to de-initialize a device"
```

Comment:
Typo: "mean" -> "means"

Reply:
Also, to respond to the question in the PR description, my initial leanings are to have it enabled by default for the very reason you gave.

Reply:
indeed, I forgot the default here no follow the current behavior

Reply:
Typo still there.

```suggestion
	bool "Provide a means to de-initialize a device"
```

Or alternatively

```suggestion
	bool "Support device de-initialization"
```